yearn/test/system/convex-strategy.test.js

```diff
@@ -0,0 +1,255 @@
+  let tbtcCurvePoolLPToken
+  let vaultDepositor
+  let tbtcCurvePoolGaugeReward
+  let tbtcCurvePoolGaugeRewardDistributor
+  let synthetixCurveRewards
+  let synthetixCurveRewardsOwner
+  let booster
+  let vault
+  let strategy
+
+  before(async () => {
+    await resetFork(12786839)
+
+    // Setup roles.
+    vaultGovernance = await ethers.getSigner(0)
+    vaultDepositor = await impersonateAccount(
+      tbtcCurvePoolLPTokenHolderAddress,
+      vaultGovernance
+    )
+    tbtcCurvePoolGaugeRewardDistributor = await impersonateAccount(
+      tbtcCurvePoolGaugeRewardDistributorAddress,
+      vaultGovernance
+    )
+    synthetixCurveRewardsOwner = await impersonateAccount(
+      synthetixCurveRewardsOwnerAddress,
+      vaultGovernance
+    )
+
+    // Get tBTC v2 Curve pool LP token handle.
+    tbtcCurvePoolLPToken = await ethers.getContractAt(
+      "IERC20",
+      tbtcCurvePoolLPTokenAddress
+    )
+
+    // Setup Synthetix staking rewards to provide Curve pool's gauge additional
+    // rewards.
+    await setupSynthetixRewards()
+
+    // Get Convex booster handle.
+    booster = await ethers.getContractAt("IConvexBooster", boosterAddress)
+
+    // Get a handle to the Yearn registry.
+    const registry = await ethers.getContractAt(
+      "IYearnRegistry",
+      registryAddress
+    )
+
+    // Always use the same vault release to avoid test failures in the future.
+    // Target release index is taken from the deployed Yearn registry contract.
+    // We aim for version 0.4.2 (see BaseStrategy.apiVersion()) whose index is 9.
+    const targetReleaseIndex = 9
+    const releaseDelta = (await registry.numReleases()) - 1 - targetReleaseIndex
+
+    // Deploy a new experimental vault accepting tBTC v2 Curve pool LP tokens.
+    const tx = await registry.newExperimentalVault(
+      tbtcCurvePoolLPToken.address,
+      vaultGovernance.address,
+      vaultGovernance.address, // set governance to be the guardian as well
+      vaultGovernance.address, // set governance to be the rewards target as well
+      vaultName,
+      vaultSymbol,
+      releaseDelta
+    )
+
+    // Get a handle to the experimental Yearn vault.
+    vault = await ethers.getContractAt(
+      "IYearnVault",
+      extractVaultAddress(await tx.wait())
+    )
+
+    // Just make sure the vault has been created properly.
+    expect(await vault.name()).to.be.equal(vaultName)
+
+    // Deploy the ConvexStrategy contract.
+    const ConvexStrategy = await ethers.getContractFactory("ConvexStrategy")
+    strategy = await ConvexStrategy.deploy(
+      vault.address,
+      tbtcCurvePoolDepositorAddress,
+      tbtcConvexRewardPoolId
+    )
+    await strategy.deployed()
+
+    // Add ConvexStrategy to the vault.
+    await vault.addStrategy(
+      strategy.address,
+      10000, // 100% debt ratio
+      0, // zero min debt per harvest
+      BigNumber.from(2).pow(256).sub(1), // infinite max debt per harvest
+      1000 // 10% performance fee
+    )
```

Comment:
Perhaps constants?

Reply:
Sure. I'll address this during the refactoring I'm planning to do at the end.

Reply:
Please see: https://github.com/keep-network/tbtc-v2/pull/10/commits/3006e09515af28c1d15e54df2095d6ddb460253e